app/poi_api/poi.py

```diff
@@ -1,12 +1,14 @@
 from datetime import datetime
 from app import common
 from app.firebase import firestore_db
+from app.poi_api.poi_errors import POINotFoundError
 from flask import jsonify
 
 
-class POI:
+class POI(common.FirebaseDataEntity):

```

Comment:
I don't think we're going in this direction. I left a similar comment on Youssef's PR but this creates tight coupling between the database and the data objects. e.g. I should be able to create a POI without a database reference. Keep the data classes for data representations and helper functions on that data. Keep the database operations in your service class.